### PR TITLE
Add MultiPolygon support for feature2 to @turf/boolean-within

### PR DESCRIPTION
**Pull request by DenisCarriere**
## Add MultiPolygon support for `feature2` to `@turf/boolean-within`

Add support for:

- Point => MultiPolygon
- MultiPoint => MultiPolygon
- LineString => MultiPolygon
- Polygon => MultiPolygon

This would also fix https://github.com/Turfjs/turf/issues/1169 by allowing the `mask` to be a MultiPolygon geometry type.

### CODE REVIEW

**Comment by DenisCarriere** [comment]
@rowanwins have a glance at these changes, they are pretty straightforward since `@turf/boolean-point-in-polygon` already supports Polygon & MultiPolygon

File: packages/turf-boolean-within/index.js [hunk 1]
```diff
@@ -34,6 +34,7 @@ function booleanWithin(feature1, feature2) {
         case 'LineString':
             return booleanPointOnLine(geom1, geom2, {ignoreEndVertices: true});
         case 'Polygon':
+        case 'MultiPolygon':
             return booleanPointInPolygon(geom1, geom2, {ignoreBoundary: true});
```
Comment (rowanwins) [inline packages/turf-boolean-within/index.js]: Don't you love easy fixes :)

**Comment by rowanwins** [comment]
Just found that this change is going to be helpful for my polygon-tangents bug #785 . Have the bug pretty much resolved, just need this to get published and it should be good to go!

File: packages/turf-boolean-within/index.js (hunk 1 above)
Comment (DenisCarriere) [reply packages/turf-boolean-within/index.js]: Very easy fix 👍 Hence why I didn't tackle `MultiLineStrings` or anything more complicated than that, lol

**Comment by DenisCarriere** [comment]
👍 Merging

**Comment by rowanwins** [comment]
Turns out I didn't need the fix for `polygonTangents` after all but Im still glad you did it :)

**Pull request merged**